### DISCUSSION TITLE
Update defaults when creating new items

### DISCUSSION THREAD

**Discussion by jmaas-onoffice** [onOffice-Web-Org/oo-wp-plugin#329] [_Discarded pitches]
# The current defaults have problems
After creation, we tend to always make the same changes. This indicates that the defaults are not good anymore.

Examples:
1. When creating a new estate list, we almost always do the following:
    - Activate the "Cover Photo"
    - Remove the field "State" which is usually deactivated in enterprise and appears red.
    - Activate "Show Estate Status"
2. When setting up the detail view for the first time, we almost always:
    - Activate all picture types.
    - Remove the field "State" which is usually deactivated in enterprise and appears red.
    - Activate "Show Estate Status"
3. When creating a new interest form, we almost always:
    - Remove street, 
    - Add search criteria like purchase price, cold rent, typeof commercialization etc.

There are some additional improvements, like setting the default amount of estates in a list to 12. On desktop we usually have three columns, while on tablet usually can only fit two. The current default of 20 is not divisible by 3, which means that the last row has only one item and thus is not filled out. By using 12, when there are 3 columns, all rows are filled out, which looks nicer.

# Update the defaults
The defaults when creating new items should be updated as follows.

There are many small changes, but they should be straight-forward default changes. So I propose this as a 1-week project.

## Estate lists
- Activate only the "Cover photo".
- Set "Records per Page" to 12.
- Activate "Show Estate Status".
- The default field list should be (`bundesland` is removed):
  - `objekttitel`
  - `objektart`
  - `objekttyp`
  - `vermarktungsart`
  - `plz`
  - `ort`
  - `objektnr_extern`
  - `wohnflaeche`
  - `grundstuecksflaeche`
  - `nutzflaeche`
  - `anzahl_zimmer`
  - `anzahl_badezimmer`
  - `kaufpreis`
  - `kaltmiete`

## Detail view
- Activate all photo types.
- Activate "Show Estate Status".
- In "Additional Media":
  - Set "Movie Links" to "Player"
  - Set "Ogulo-Links" to "Embedded"
  - The PDF-Expose should remain empty, and the other two link types should remain "Deactivated".
- The default field list should be (`bundesland` is removed, some fields for the "Energieausweis" and comission are added at the end):
  - `objekttitel`
  - `objektart`
  - `objekttyp`
  - `vermarktungsart`
  - `plz`
  - `ort`
  - `objektnr_extern`
  - `wohnflaeche`
  - `grundstuecksflaeche`
  - `nutzflaeche`
  - `anzahl_zimmer`
  - `anzahl_badezimmer`
  - `kaufpreis`
  - `kaltmiete`
  - `objektbeschreibung`
  - `lage`
  - `ausstatt_beschr`
  - `sonstige_angaben`
  - `baujahr`
  - `endenergiebedarf`
  - `energieverbrauchskennwert`
  - `energieausweistyp`
  - `energieausweis_gueltig_bis`
  - `energyClass`
  - `aussen_courtage`
  - `kaution`

## Similar estates
- Set "Amount of Estates" to 6.

## Unit lists
- Activate only the "Cover photo".
- The default field list should be (some fields are removed):
  - `objekttitel`
  - `objektart`
  - `objekttyp`
  - `vermarktungsart`
  - `plz`
  - `ort`
  - `objektnr_extern`
  - `wohnflaeche`
  - `grundstuecksflaeche`
  - `nutzflaeche`
  - `anzahl_zimmer`
  - `anzahl_badezimmer`
  - `kaufpreis`
  - `kaltmiete`

## Contact form
- The default field list should be (address fields are removed):
  - `Vorname`
  - `Name`
  - `Telefon1`
  - `Email`
  - `message`

## Interest form
- The default field list should be (address fields are removed, some search criteria are added):
  - `Vorname`
  - `Name`
  - `Telefon1`
  - `Email`
  - `vermarktungsart`
  - `objekttyp`
  - `kaufpreis`
  - `kaltmiete`
  - `wohnflaeche`
  - `anzahl_zimmer`

## Owner form
The owner form's defaults should stay the same.

## Applicant search form
- The default field list should be (address fields are removed, some search criteria are added):
  - `vermarktungsart`
  - `objekttyp`
  - `kaufpreis`
  - `kaltmiete`
  - `wohnflaeche`
  - `anzahl_zimmer`

## Address list
- The default field list should be (it is currently empty):
  - `Anrede`
  - `Vorname`
  - `Name`
  - `Zusatz1`
  - `Email`
  - `Telefon1`
  - `Telefax1`

# Won't dos
There are a few related issues that we should separate into their own bets, to keep this one a small-batch bet.
- When price on request is active for an estate, the price is still shown.
- The GDPR field is output as a select instead of a checkbox.
- In the interest form, currently both "purchase price" and "cold rent" are always visible. When the user selects "Rent", the purchase price should be hidden.

**Comment by jmaas-onoffice** [comment]
Originally, you asked for "Record per Page" of Similar Estates to be set to 12, but it currently looks like this:
![grafik](https://user-images.githubusercontent.com/90763364/190375181-1db8a195-8b02-47d3-be21-b8ea5d3840e8.png)

I changed the bet to set "Amount of Estates" to 6.

**Comment by jmaas-onoffice** [comment]
For the applicant search form, you asked to remove address fields (street, zip code, city) and to add some fields (construction year, condition) which are not available for this form.

Therefore I simply copied the search criteria from the interest form.

**Comment by jmaas-onoffice** [comment]
For the interest form, you asked to add "Geo-Position" for "Radius search fields". What does that mean?

**Comment by jmaas-onoffice** [comment]
For the address list, you asked for "Position". This seems to be a field that is hidden behind a tab:
![grafik](https://user-images.githubusercontent.com/90763364/190383372-db102e79-c596-482d-9805-05ccef27ba75.png)

This made me question whether it is used regularly, and I decided to remove it.

Is this a field that should be added back?

**Comment by dai-eastgate** [comment]
@jmaas-onoffice 
![image](https://user-images.githubusercontent.com/106214469/192473694-b0c0683c-8660-4568-ba0a-ca7d67b59458.png)
In detail view, do you want to remove " bundesland"?

**Reply by jmaas-onoffice** [reply]
Yes, well spotted! I've fixed the description, thanks!
